### PR TITLE
Fix Incorrect Persistence of Multiproperties in FulgoraGraphComputer #943

### LINKED ISSUE #943: VertexProgram Fails to Persist Multiproperties (FulgoraGraphComputer)
When setting a multiproperty in the execute() method of a `VertexProgram`, it updates an `ArrayList` with new values of the property.  This `ArrayList` is supposed to be iterated over when persisting results to the graph in` FulgoraGraphComputer.submit()`.  However, when performing a persist to the original graph (`ResultGraph.ORIGINAL`), it attempts to write the `ArrayList` as a single property, which fails since JanusGraph does not provide a default `ArrayList` serializer.  I have code demonstrating this bug in https://github.com/bgamer50/janusgraph-test.

The fix for this is to check if an `ArrayList` is being written, and if so, write each individual value using `forEach()` using the predefined JanusGraph schema (either `Cardinality.LIST` or `Cardinality.SET`).  If the JanusGraph schema was not defined as one of those two, throw an Exception.  I have code for this fix in https://github.com/bgamer50/janusgraph-ab/tree/fix-vp-persist (`FulgoraGraphComputer` class).

### PR DESCRIPTION
**Pull request by bgamer50**
Fixes #943 
Signed-off-by: bgamer50 <bgamer50@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by bgamer50** [comment]
@amcp I renamed the pull request and modified the previous commit.  Unfortunately it looks like it has to run the CI again.  It should be ready after that.  Are there any other issues I should address?

**Comment by amcp** [comment]
@bgamer50 please get a CCLA or ICLA on file. I couldn't find one in JanusGraph/legal, so you need to get one on file.

**Comment by bgamer50** [comment]
@amcp I just resubmitted the ICLA per the instructions in janusgraph/legal.  Not sure how frequently it gets updated.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java [hunk 1]
```diff
@@ -327,13 +329,25 @@ public GraphComputer mapReduce(final MapReduce mapReduce) {
...
                     for (Map.Entry<Long, Map<String, Object>> vertexProperty : mutatedProperties.entrySet()) {
                         Vertex v = resultgraph.vertices(vertexProperty.getKey()).next();
                         for (Map.Entry<String, Object> prop : vertexProperty.getValue().entrySet()) {
                             if (prop.getValue() instanceof List) {
-                                ((List) prop.getValue())
-                                    .forEach(value -> v.property(VertexProperty.Cardinality.list, prop.getKey(), value));
+                                ((List<Object>) prop.getValue())
+                                    .forEach(p -> {
+                                    	Cardinality cardinality = tx.getPropertyKey(prop.getKey()).cardinality();
+	                        			if(cardinality.equals(Cardinality.LIST)) {
```
Comment (analytically) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java]: spacing

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java [hunk 2]
```diff
@@ -327,13 +329,25 @@ public GraphComputer mapReduce(final MapReduce mapReduce) {
...
+	                        				v.property(VertexProperty.Cardinality.list, prop.getKey(), p);
+	                        			}
+	                        			else if(cardinality.equals(Cardinality.SET)) {
```
Comment (analytically) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/olap/computer/FulgoraGraphComputer.java]: spacing

**Pull request closed without merging**